### ISSUE TITLE
Breaks on temporary tables.

### ISSUE THREAD

**Issue by coreyatmoat** [keithf4/pg_bloat_check#2]
Ran the script during an ETL which creates lots of temporary tables and got this error immediately:

```
Traceback (most recent call last):
  File "pg_bloat_check.py", line 357, in <module>
    get_bloat(conn, tuple(exclude_schema_list), tuple(include_schema_list), exclude_object_list)
  File "pg_bloat_check.py", line 218, in get_bloat
    cur.execute(sql, [o['oid'], args.min_size, args.min_wasted_size, args.min_wasted_percentage])
  File "/usr/local/lib/python2.7/dist-packages/psycopg2/extras.py", line 120, in execute
    return super(DictCursor, self).execute(query, vars)
psycopg2.NotSupportedError: cannot access temporary tables of other sessions
```

**Comment by coreyatmoat** [comment]
maybe filter pg_class on relpersistence <> 't', because we don't care bloat in temp tables regardless of who owns them? Testing a patch for that now. Might be a while because it's a big db.

**Comment by keithf4** [comment]
I'll wait a bit for your patch, otherwise that's an easy fix. Thanks for all the testing!

**Comment by coreyatmoat** [comment]
My patched script and the daily ETL might be having a locking slapfight right now, so rather than waiting for it to finish, here's the _very_ simple patch
```
116c116,117
<                     WHERE relkind IN ('r', 'm') """
---
>                     WHERE relkind IN ('r', 'm')
>                     AND c.relpersistence <> 't' """
124c125,126
<                     AND a.amname <> 'gin' """
---
>                     AND a.amname <> 'gin'
>                     AND c.relpersistence <> 't' """
```

**Comment by keithf4** [comment]
Fixed this in v2.0.2. Looked through pg_class & pg_index to see if there was anything else. Also fixed it so it skips indexes that are marked as being dropped.

**Comment by EricVeldhuyzen** [comment]
I just noticed that this still breaks on temporary indexes, I was able to fix this with a very similar patch to above:

``
     sql_indexes = """ SELECT c.oid, c.relkind, c.relname, n.nspname, i.indisprimary, c.reloptions 
                      FROM pg_catalog.pg_class c
                      JOIN pg_catalog.pg_namespace n ON c.relnamespace = n.oid
                      JOIN pg_catalog.pg_index i ON c.oid = i.indexrelid
                      JOIN pg_catalog.pg_am a ON c.relam = a.oid
                      WHERE c.relkind = 'i' 
+                     AND c.relpersistence <> 't'
                      AND a.amname <> 'gin' AND a.amname <> 'brin' """
  
  
      cur.execute("SELECT current_setting('server_version_num')::int >= 90300")
      if cur.fetchone()[0] == True:
          sql_indexes += " AND indislive = 'true' "
``

**Comment by keithf4** [comment]
@EricVeldhuyzen Thanks for finding that and including a fix. I've pushed up an update for v2.6.2 that includes this. Mind giving it a quick test and if all looks good I will tag the new version.

**Comment by EricVeldhuyzen** [comment]
Thanks, I already patched my 2.6.1 version in production with my fix, so yes, this solves this problem.
